api/tests/opentrons/protocol_engine/execution/test_equipment_handler.py

```
@@ -1,15 +1,16 @@
 """Test equipment command execution side effects."""
 
-import pytest
-from _pytest.fixtures import SubRequest
+from unittest.mock import sentinel
```

mmmm are we doing this differently now?

---

Are you referring to the introduction of `unittest.mock.sentinel`?

Yes, [I advocate for it.](https://opentrons.slack.com/archives/CTLS4MBB4/p1711493303201389)

As you know, in the Decoy doctrine, we have "collaborator functions" that, ideally, do nothing except pass values between dependencies. Collaborators should treat those values as opaque and not implement any "real logic" themselves.

That is *exactly* what Python's standard `unittest.mock.sentinel` feature is for. And even though it comes from `unittest.mock`, there's no reason we can't use it with `decoy`.

Not having to construct so many dummy objects does a lot to [cut tests down](https://github.com/Opentrons/opentrons/commit/0f86788a96a08315365b8f19bfe60081a64154e5) and make them more readable and maintainable.

---

I agree. this is really cool and clean. I will try it myself :-) still need to see how this plays.